tanner/emulators/base.py

```diff
@@ -23,57 +15,66 @@ def __init__(self, base_dir, db_name, loop=None):
             'sqli': sqli.SqliEmulator(db_name, base_dir),
             'cmd_exec': cmd_exec.CmdExecEmulator()
         }
+        self.get_emulators = ['sqli', 'rfi', 'lfi', 'xss', 'cmd_exec']
+        self.post_emulators = ['sqli', 'rfi', 'lfi', 'xss', 'cmd_exec']
 
-    async def handle_post(self, session, data):
+    def extract_get_data(self, path):
+        """
+        Return all the GET parameter
+        :param path: The URL path from which GET parameters are to be extracted
+        :return: A MultiDictProxy object containg name and value of parameters
+        """
+        path = urllib.parse.unquote(path)
+        encodings = [('&&', '%26%26'), (';', '%3B')] 
+        for value, encoded_value in encodings:
+            path = path.replace(value, encoded_value)
+        get_data = yarl.URL(path).query
+        return get_data
+
+    async def get_emulation_result(self, session, data, target_emulators):
+        """
+        Return emulation result for the vulnerabilty of highest order
+        :param session: Current active session
```

Comment:
I think in this case params of the function is completely understandable from the context. Do we really need to write that session is session? :) I think we should write more self-documented code than more docstrings. How do you think?